immix/llvm/plimmix.cpp

```diff
@@ -30,9 +30,68 @@ extern "C" void LLVMLinkPLImmixGC()
 #include "llvm-c/Transforms/PassManagerBuilder.h"
 
 extern "C" void add_module_pass(llvm::legacy::PassManagerBase * PB) {
-    PB->add(new Immix());
+    PB->add(new ImmixLegacy());
+
+}
+#include "llvm/Passes/PassBuilder.h"
+#include "llvm/Passes/PassPlugin.h"
+
+#include "llvm-c/Types.h"
+#include "llvm-c/Transforms/PassBuilder.h"
+
+// param: opt opt level
+extern "C" void run_module_pass(LLVMModuleRef  M, int opt) {
```

Comment:
could add a comment about `run_module_pass` to explain:
- what does this function do
- which llvm API it calls(add a llvm link if possible)
- the mapping between optimization level and integer.